### PR TITLE
Mekhanik evgenii/fix 1825

### PR DESCRIPTION
**Pull request by EvgeniiMekhanik**

### CODE REVIEW

File: fw/http_frame.c [hunk 1]
```diff
@@ -73,19 +73,6 @@ typedef enum {
 	HTTP2_RECV_APP_DATA_POST
```
Comment (krizhanovsky) [inline fw/http_frame.c:73]: Please update the copyright year of all the updated files and/or initialize the git hook, which does it automatically

File: fw/http_frame.c [hunk 2]
```diff
@@ -1006,13 +996,21 @@ tfw_h2_apply_settings_entry(TfwH2Ctx *ctx, unsigned short id,
...
 	       ctx->hdr.stream_id, ctx->hdr.flags);
 
-	ctx->hpack.max_window = ctx->lsettings.hdr_tbl_sz;
-	/*
-	 * TODO: apply other local settings on ACK receiving.
-	 */
+	CHECK_AND_APPLY_SETTING(HTTP2_SETTINGS_TABLE_SIZE, {
+		ctx->hpack.max_window = ctx->lsettings.hdr_tbl_sz;
+		ctx->hpack.dec_tbl.wnd_update = true;
+	})
```
Comment (krizhanovsky) [inline fw/http_frame.c:1013]: Single usage macro just makes the code more sophisticated without a reason

**Review by krizhanovsky — APPROVED** [review]
LGTM, but a good review from @const-t is required since I'm not so confident with the HPACK code

File: fw/http.c [hunk 3]
```diff
@@ -5814,14 +5826,19 @@ tfw_http_req_process(TfwConn *conn, TfwStream *stream, struct sk_buff *skb,
...
 		 */
+	case T_COMPRESSION:
+		fallthrough;
 	case T_BAD:
 		T_DBG2("Drop invalid HTTP request\n");
 		TFW_INC_STAT_BH(clnt.msgs_parserr);
-		return tfw_http_req_parse_drop_with_fin(req, 400, NULL);
+		return tfw_http_req_parse_drop_with_fin(req, 400, NULL, 
+							r == T_COMPRESSION ?
+							HTTP2_ECODE_COMPRESSION : HTTP2_ECODE_PROTO);
```
[the remark this answers is not in the document]
Comment (EvgeniiMekhanik) [reply fw/http.c]: Yes fixed

**Pull request merged**